validate.py

```diff
@@ -175,7 +178,10 @@ def check_registration(old, new, op):
             #no destroying the day-of object
             'day_of': say_no_to_non_admin,
             'day_of\\.[A-Za-z1-2_]+': say_no_to_non_admin,
-            'registration_status': check_registration
+            'registration_status': check_registration,
+            #auth tokens are never given access. auth is most important token, but 'token' is added as a bonus safety measure
+            'auth': say_no,
+            'token': say_no
```

Comment:
What is `token`? How is this field present in the user.

https://docs.mongodb.com/manual/tutorial/query-arrays/#query-for-an-element-by-the-array-index-position suggests that the regex would be `auth\\.[0-9]+\\.token` but you'll notice that it's caught by `auth` as a regex already.